### DISCUSSION TITLE
Help with tiling images

### DISCUSSION THREAD

**Discussion by gingerninja2000** [ImageMagick/ImageMagick#4827] [Help]
Hi

I have a folder full of seamless tile images, I want to tile all the files in the folder into 3x3. I just want a simple command to run to tile and save over the existing files.

Please can someone help me?

**Comment by fmw42** [comment]
What OS?  What version of ImageMagick?  Are you asking to create 3x3 array for each image so that your images are larger?  You could just loop over each image and use the montage command to create a 3x3 tiling of for each image. See https://imagemagick.org/script/montage.php

**Reply by gingerninja2000** [reply]
windows 10, latest version, I just installed it today, can you help me with the full command line. I want it to create a 3x3 of all the files in one folder, it can overwrite the file or create new ones, I just need to create a set of these 3x3 images Thanks Alex

**Reply by gingerninja2000** [reply]
I'm creating tile images for printing onto fabric, I want to show the client how the tiles will look when in an array

**Comment by fmw42** [comment]
Sorry, I do not script in Windows. You have to write a script or .bat file to loop over each image in your directory. Then just run montage to make the tile. 

What is your ImageMagick version?  What did you install?

**Comment by gingerninja2000** [comment]
ImageMagick-7.1.0-23-Q16-HDRI-x64-dll.exe

I don't know how to do that, that's why I'm here asking for help. I've tried reading the help but I cant figure it out.

**Comment by fmw42** [comment]
Given some image.suffix (put your image filename and suffix in there), then this will overwrite your input with the tiled image in a 3x3 array

`magick montage image.suffix -duplicate 8 -tile 3x3 -geometry +0+0 image.suffix`

Example (here I change the output name slightly)

![chalk_seamless_tile](https://user-images.githubusercontent.com/13458218/153329714-48c5bd2f-7bcc-4dd7-b05d-cb0ee6b20b55.png)

`magick montage chalk_seamless_tile.png -duplicate 8 -tile 3x3 -geometry +0+0 chalk_seamless_tile_3x3.png`

![chalk_seamless_tile_3x3](https://user-images.githubusercontent.com/13458218/153329878-187bd9d0-fd3a-404d-8f7c-30d1cba1152d.png)

**Comment by gingerninja2000** [comment]
Fred

Thanks, I got it working (sort of). How can I make it apply the same script
to all files in a folder and add a suffix to the filename?

Alex

**Comment by fmw42** [comment]
You have to write a script loop over each image in your folder and then use montage for each.  I could tell you how to do that in Unix, but sorry, I do not know how to script on Windows. Perhaps one of the other IM users might be able to help.

**Comment by snibgo** [comment]
I suggest using a Windows "for" loop. Something like this (untested):
```
for %F in (*.png) do magick montage %F -duplicate 8 -tile 3x3 -geometry +0+0 newdir\%F
```
If the command is in a BAT script file, double the % signs: "%%F". If your filenames contain spaces, you should quote them: "%F".

For help on "for", type "help for" at the command line.

EDIT: corrected the "for" part of the command.

**Reply by gingerninja2000** [reply]
hi, I got this to work if I dont have spaces in the filenames, I tried replacing %F with "%F" but didnt work, can you help me with an example please?

**Reply by fmw42** [reply]
Put double quotes about your file names that have spaces in them or on all file names.

**Reply by gingerninja2000** [reply]
thanks I just removed the spaces in the filenames, I didn't understand fully what you meant, thanks for your help.

**Reply by fmw42** [reply]
When you have spaces in name, the system thinks there are more than one file. To make sure the system thinks there is only one file with spaces in the name, put double quotes about the file name. It won't hurt if you put double spaces about all filenames. For example in place of `my test image.png` use `"my test image.png"`

**Comment by gingerninja2000** [comment]
Ok, thanks.

**Comment by gingerninja2000** [comment]
How do I edit the command to insert a "_3x3" in the filename before the
.png?

**Reply by fmw42** [reply]
Use a text editor. On Windows, see Notepad or WordPad.

**Reply by gingerninja2000** [reply]
I mean how should it look? If I do it like this below it puts the _3x3 after .png, sorry I'm new to all of this

for %F in (*.png) do magick montage %F -duplicate 8 -tile 3x3 -geometry +0+0 newdir\%F_3x3

**Reply by fmw42** [reply]
snibgo will have to help you. You will need separate the suffix from the rest of the file name and put the _3x3 just before adding the suffix back.

**Comment by snibgo** [comment]
@gingerninja2000: Have you tried my suggestion?
```
For help on "for", type "help for" at the command line.
```
Towards the end of that, it shows how to use "%~n". For example:
```
for %F in (*.png) do magick montage "%F" -duplicate 8 -tile 3x3 -geometry +0+0 "newdir\%~nF_3x3.png"
```
The subdirectory "newdir\" must already exist. Use whatever path you want.

If you don't want to use a different directory for the outputs then remove "newdir\\", obviously. But then if you re-run the command you will get 3x3 versions of the 3x3 versions.

EDIT: If you have spaces in your filenames or the output directory, then you need quotes, as I have edited into this sample command.

**Reply by gingerninja2000** [reply]
thanks this is really helpful
